### PR TITLE
run docker build on all branches, but only deploy on main

### PR DESCRIPTION
**Pull request by exstral**
Run the docker build on all pushes so we can be sure the dependencies are okay before we merge a pull request :)

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -21,23 +20,32 @@ jobs:
     steps:
       - uses: actions/checkout@v1
 
-      - name: Extract branch name
+      - name: Slugify branch name for use in docker image tags
         shell: bash
-        run: echo "##[set-output name=branch;]$(echo ${GITHUB_REF#refs/heads/})"
-        id: extract_branch
+        run: echo "##[set-output name=slug;]$(echo $GITHUB_REF | sed -E 's/refs\/(heads|tags)\///g' | iconv -t ascii//TRANSLIT | sed -E 's/[^a-zA-Z0-9]+/-/g' | sed -E 's/^-+|-+$//g' | tr A-Z a-z)"
```
Comment (lwille) [inline .github/workflows/docker.yml]: :exploding_head:

Comment (lwille) [reply .github/workflows/docker.yml]: this is only required for cases where we want to push the image, right?

Comment (exstral) [reply .github/workflows/docker.yml]: haha yes. I have this setup in my day job where I push the image to separate tags based on the branch. So if you edit the docker setup on a branch, it shouldn't break the image used in another branch. Currently unused here cause we only push `main`.

But I still like to keep it around cause it took a long time to find this one liner that works and we might want to do it in the future :innocent:

Comment (exstral) [reply .github/workflows/docker.yml]: Changed it so that it only slugs the branch name when it actually pushes, on main branch.

**Review by lwille — APPROVED** [review]

**Pull request merged**